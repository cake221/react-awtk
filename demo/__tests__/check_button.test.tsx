import "../check_button"

describe("test check_button", ()=>{

  it("toMatchSnapshot", ()=>{
    expect(__AwtkSnapshot).toMatchSnapshot()
  })
})
