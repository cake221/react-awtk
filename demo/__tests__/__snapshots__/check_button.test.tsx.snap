// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test check_button toMatchSnapshot 1`] = `
Array [
  "getRootHostContext",
  "getChildHostContext",
  "shouldSetTextContent",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :function( parent, x, y, w, h ){
  global.__AwtkSnapshot.push(\`funtion  check_button_create( parent, x, y, w, h ) :\${parent}, \${x}, \${y}, \${w}, \${h}\`)
  return undefined
}, 10, 10, 128, 30",
  "funtion  widget_set_children_layout( widget, params ) :function( parent, x, y, w, h ){
  global.__AwtkSnapshot.push(\`funtion  check_button_create( parent, x, y, w, h ) :\${parent}, \${x}, \${y}, \${w}, \${h}\`)
  return undefined
}, ",
  "funtion  widget_on( widget, type, on_event, ctx ) :function( parent, x, y, w, h ){
  global.__AwtkSnapshot.push(\`funtion  check_button_create( parent, x, y, w, h ) :\${parent}, \${x}, \${y}, \${w}, \${h}\`)
  return undefined
}, EVT_VALUE_CHANGED, function (evt) {
                    // TODO: 需要检测是否正确
                    var btn = src_1.TCheckButton.cast(evt.target);
                    console.log(btn.name + ' changed. ' + (btn.value));
                    return TRet.OK;
                }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :function( parent, x, y, w, h ){
  global.__AwtkSnapshot.push(\`funtion  check_button_create( parent, x, y, w, h ) :\${parent}, \${x}, \${y}, \${w}, \${h}\`)
  return undefined
}, 10, 50, 128, 30",
  "funtion  widget_set_children_layout( widget, params ) :function( parent, x, y, w, h ){
  global.__AwtkSnapshot.push(\`funtion  check_button_create( parent, x, y, w, h ) :\${parent}, \${x}, \${y}, \${w}, \${h}\`)
  return undefined
}, ",
  "funtion  widget_on( widget, type, on_event, ctx ) :function( parent, x, y, w, h ){
  global.__AwtkSnapshot.push(\`funtion  check_button_create( parent, x, y, w, h ) :\${parent}, \${x}, \${y}, \${w}, \${h}\`)
  return undefined
}, EVT_VALUE_CHANGED, function (evt) {
                    var btn = src_1.TCheckButton.cast(evt.target);
                    console.log(btn.name + ' changed. ' + (btn.value));
                    return TRet.OK;
                }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :function( parent, x, y, w, h ){
  global.__AwtkSnapshot.push(\`funtion  check_button_create( parent, x, y, w, h ) :\${parent}, \${x}, \${y}, \${w}, \${h}\`)
  return undefined
}, 10, 90, 128, 30",
  "funtion  widget_set_children_layout( widget, params ) :function( parent, x, y, w, h ){
  global.__AwtkSnapshot.push(\`funtion  check_button_create( parent, x, y, w, h ) :\${parent}, \${x}, \${y}, \${w}, \${h}\`)
  return undefined
}, ",
  "funtion  widget_on( widget, type, on_event, ctx ) :function( parent, x, y, w, h ){
  global.__AwtkSnapshot.push(\`funtion  check_button_create( parent, x, y, w, h ) :\${parent}, \${x}, \${y}, \${w}, \${h}\`)
  return undefined
}, EVT_VALUE_CHANGED, function (evt) {
                    var btn = src_1.TCheckButton.cast(evt.target);
                    console.log(btn.name + ' changed. ' + (btn.value));
                    return TRet.OK;
                }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  check_button_create_radio( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 10+128, 128, 30",
  "funtion  widget_set_children_layout( widget, params ) :undefined, ",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
                    var btn = src_1.TCheckButton.cast(evt.target);
                    console.log(btn.name + ' changed. ' + (btn.value));
                    return TRet.OK;
                }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  check_button_create_radio( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 50+128, 128, 30",
  "funtion  widget_set_children_layout( widget, params ) :undefined, ",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
                    var btn = src_1.TCheckButton.cast(evt.target);
                    console.log(btn.name + ' changed. ' + (btn.value));
                    return TRet.OK;
                }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  check_button_create_radio( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 90+129, 128, 30",
  "funtion  widget_set_children_layout( widget, params ) :undefined, ",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
                    var btn = src_1.TCheckButton.cast(evt.target);
                    console.log(btn.name + ' changed. ' + (btn.value));
                    return TRet.OK;
                }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  check_button_create_radio( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 130+129, 128, 30",
  "funtion  widget_set_children_layout( widget, params ) :undefined, ",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
                    var btn = src_1.TCheckButton.cast(evt.target);
                    console.log(btn.name + ' changed. ' + (btn.value));
                    return TRet.OK;
                }, null",
  "finalizeInitialChildren",
  "createInstance",
  "funtion  window_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "finalizeInitialChildren",
  "prepareForCommit",
  "appendNodeChild",
  "resetAfterCommit",
  "getPublicInstance",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_layout( widget ) :undefined",
]
`;
